web/packages/shared/hooks/useRefClickOutside/useRefClickOutside.ts

```diff
@@ -18,6 +18,9 @@
 
 import { MutableRefObject, useEffect, useRef, useCallback } from 'react';
 
+// IGNORE_CLICK_CLASSNAME is the className that should be on elements which shouldn't trigger setOpen(false).
+export const IGNORE_CLICK_CLASSNAME = 'ignore-click';
```

Comment:
It's more the other way around, `useRefClickOutside` will close the dropdown when anything _other_ than the ref (or its contents) is clicked. This works perfectly currently, however it just doesn't handle the case where we might want to be able to click on an element outside of the ref without triggering the close. If we want to do so, then all we need to do is add this `className` to that element. I can't think of any other exceptions we would ever need to add and manage, as that would go beyond the scope of this hook altogether.

I added this because I was running into an issue with the notification list where if you clicked on the popover for a notification like "Hide this notification", it would close the whole notification list behind it. Clicking on the dialog which contains the contents of a text notification would also trigger the close. Adding this `className` to those components prevented this bug.

> It seems to me that we could use a different approach: don't close the dropdown by default and pass a close callback to children so they can do this when they need to.

This isn't a bad idea however this hook is currently being used in 7 places and this would require us to refactor its usage everywhere, which I think would be beyond the scope of this PR

Reply:
Good catch, this was being caused by two issues: First due to the fact that the element being clicked in this scenario was the menu's "Backdrop" element, which didn't have the `IGNORE_CLASS_CLASSNAME`. As well as the fact that this was being registered as a click on the notification. I've fixed both issues.